ngraph/frontend/paddlepaddle/src/frontend.cpp

```diff
@@ -155,41 +159,157 @@ namespace ngraph
+            if (is_type<VariantWrapper<std::string>>(variants[0]))
+            {
+                std::string suffix = ".pdmodel";
+                std::string model_path =
+                    as_type_ptr<VariantWrapper<std::string>>(variants[0])->get();
+                if (!pdpd::endsWith(model_path, suffix))
+                {
+                    model_path += pdpd::get_path_sep<char>() + "__model__";
+                }
+                std::ifstream model_str(model_path, std::ios::in | std::ifstream::binary);
+                // It is possible to validate here that protobuf can read model from the stream,
+                // but it will complicate the check, while it should be as quick as possible
+                return model_str && model_str.is_open();
+            }
+#if defined(ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+            else if (is_type<VariantWrapper<std::wstring>>(variants[0]))
             {
-                // The case when folder with __model__ and weight files is provided or .pdmodel file
-                return std::make_shared<InputModelPDPD>(paths[0]);
+                std::wstring suffix = L".pdmodel";
+                std::wstring model_path =
+                    as_type_ptr<VariantWrapper<std::wstring>>(variants[0])->get();
+                if (!pdpd::endsWith(model_path, suffix))
+                {
+                    model_path += pdpd::get_path_sep<wchar_t>() + L"__model__";
+                }
+                std::ifstream model_str(model_path, std::ios::in | std::ifstream::binary);
+                // It is possible to validate here that protobuf can read model from the stream,
+                // but it will complicate the check, while it should be as quick as possible
+                return model_str && model_str.is_open();
             }
-            else if (paths.size() == 2)
+#endif
+            else if (is_type<VariantWrapper<std::shared_ptr<std::istream>>>(variants[0]))
             {
-                // The case when .pdmodel and .pdparams files are provided
-                std::ifstream model_stream(paths[0], std::ios::in | std::ifstream::binary);
-                FRONT_END_INITIALIZATION_CHECK(model_stream && model_stream.is_open(),
-                                               "Cannot open model file.");
-                std::ifstream weights_stream(paths[1], std::ios::in | std::ifstream::binary);
-                FRONT_END_INITIALIZATION_CHECK(weights_stream && weights_stream.is_open(),
-                                               "Cannot open weights file.");
-                return load_from_streams({&model_stream, &weights_stream});
+                // Validating first stream, it must contain a model
+                std::shared_ptr<std::istream> p_model_stream =
+                    as_type_ptr<VariantWrapper<std::shared_ptr<std::istream>>>(variants[0])->get();
+                paddle::framework::proto::ProgramDesc fw;
```

Comment:
Maybe can we generalize approach and use `ParseFromIstream` for any arguments?

Reply:
It would have a drawback on performance